src/main/java/seedu/address/logic/commands/RequiredCommand.java

```diff
@@ -0,0 +1,319 @@
+     * any undone Science Modules.
+     */
+    public void compareScience() {
+        boolean add = true;
+        for (Module module : requiredScience) {
+            for (Module mod : modules) {
+                if (module.isSameModule(mod)) {
+                    add = false;
+                    break;
+                }
+            } if (!add) {
+                break;
+            }
+        } if (add) {
+            leftOverModules += MESSAGE_SCIENCE + "\n";
+        } else {
+            leftOverModules += MESSAGE_SUCCESS_SCIENCE + "\n";
+        }
+        leftOverModules += "\n";
+    }
+
+    /**
+     * Returns internshipStorage attribute of RequiredCommand object.
+     * @return internshipStorage attribute of type Optional<ReadOnlyGradPad/>.
+     */
+    public Optional<ReadOnlyGradPad> getInternshipStorage() {
+        return internshipStorage;
+    }
+
+    /**
+     * Loads the internshipStorage attribute with Internship Modules.
+     * @throws IOException
+     * @throws DataConversionException
+     */
+    public void setInternshipStorage(Path path) throws IOException, DataConversionException {
+        JsonGradPadStorage storage = new JsonGradPadStorage(path);
+        internshipStorage = storage.readGradPad();
+    }
+
+    /**
+     * Cross references the user's current list of Modules and marks out
+     * any undone Internship Modules. Also calculates current MC score
+     * achieved from Internship Modules.
+     */
+    public void compareInternship() {
+        int modularScore = 0;
+        String leftOverInternship = "";
+        for (Module module : requiredInternship) {
+            boolean add = true;
+            for (Module mod : modules) {
+                if (module.isSameModule(mod)) {
+                    int modularCredits = Integer.parseInt(mod.getModularCredits().toString());
+                    modularScore += modularCredits;
+                    add = false;
+                    break;
+                }
+            } if (add) {
+                String moduleToAdd = module.getModuleCode() + " (" + module.getModularCredits() + " MCs)";
+                leftOverInternship += "\n" + moduleToAdd;
+            }
+        } if (modularScore < 12) {
+            String modScore = " You are currently at " + modularScore + " MCs. ";
+            leftOverModules += "\n" + MESSAGE_INTERN_1 + modScore + MESSAGE_INTERN_2 + leftOverInternship;
+        } else {
+            leftOverModules += MESSAGE_SUCCESS_INTERN;
+        }
+    }
+    @Override
+    public CommandResult execute(Model model) throws IOException, DataConversionException {
+        requireNonNull(model);
+        modules = model.getGradPad().getModuleList();
+        setFoundationStorage(FOUNDATION_PATH);
+        setITprofStorage(ITPROF_PATH);
+        setMathAndScienceStorage(MATHANDSCI_PATH);
+        setScienceStorage(SCIENCE_PATH);
+        setInternshipStorage(INTERNSHIP_PATH);
```

Comment:
(cont. from prev comment)
So maybe what we can is to wrap these error-throwing calls in a try-catch block and handle exceptions here instead. After all, clients expect this `execute` method to give them a result (success or failure message), and not an exception.

```suggestion
try {
        setFoundationStorage(FOUNDATION_PATH);
        setITprofStorage(ITPROF_PATH);
        setMathAndScienceStorage(MATHANDSCI_PATH);
        setScienceStorage(SCIENCE_PATH);
        setInternshipStorage(INTERNSHIP_PATH);
} catch (IOException | DataConversionException e) {
        // from the error msg you wrote in MainWindow hehe
        return new CommandResult("File not found :(");
}
```